### DISCUSSION TITLE
Complete setup to train model and use with esp32 as the receiver.

### DISCUSSION THREAD

**Discussion by KaushikNITK** [ruvnet/RuView#224] [Q&A]
I want to set this up to detect keypoints of a person. I will be training this model maybe finetune.  I want to make sure few things

1. Is it compulsory to have the person in line of sight?
2. What happens if metal is in the line of sight ( person is in line of sight and when person is not in line of sight)
3. What happens if there are electronic components near transmitter or receiver
4. How does it make sure that if the user places transmitter at a different position than the few with which it is trained on.
5. can i get trained weights of the models for python

I want to know more about these kind of questions.

**Comment by ruvnet** [comment]
Great questions:

**1. Line of sight required?** No — WiFi CSI works through walls. Accuracy decreases with more obstacles but drywall/wood are fine.

**2. Metal in line of sight?** Metal reflects WiFi strongly. Static metal is OK (system calibrates around it). Moving metal objects will look like human movement — known limitation.

**3. Electronics near TX/RX?** Usually fine. Keep ESP32 nodes 30cm+ from routers and large metal appliances.

**Training workflow:**
1. Flash ESP32-S3 nodes with [v0.4.3.1](https://github.com/ruvnet/RuView/releases/tag/v0.4.3.1-esp32)
2. Record CSI data via sensing server API
3. Label recordings with ground-truth poses
4. Train with `wifi-densepose-train` crate

See the [user guide training section](https://github.com/ruvnet/RuView/blob/main/docs/user-guide.md#training-a-model).
